### PR TITLE
Refactor CellRegistrator

### PR DESCRIPTION
**Pull request by asirokova**

### CODE REVIEW

File: ImagePicker/CellRegistrator.swift [hunk 1]
```diff
@@ -20,28 +20,26 @@ import Photos
...
     }
     
-    // MARK: Private Methods
-    
-    fileprivate let actionItemIdentifierPrefix = "eu.inloop.action-item.cell-id"
-    fileprivate var actionItemNibsData: [Int: (UINib, String)]?
-    fileprivate var actionItemClassesData: [Int: (UICollectionViewCell.Type, String)]?
+    // MARK: - Private Methods
+    private let actionItemIdentifierPrefix = "eu.inloop.action-item.cell-id"
+    internal var actionItemNibsData: [Int: (UINib, String)]?
```
Comment (radimhalfar) [inline ImagePicker/CellRegistrator.swift]: why to specify internal? It is marked internal by default if unspecified

File: ImagePicker/UICollectionView+CellRegistrator.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
+//
+//  UICollectionView+CellRegistrator.swift
+//  ImagePicker
+//
+//  Created by Anna Shirokova on 14/06/2018.
```
Comment (radimhalfar) [inline ImagePicker/UICollectionView+CellRegistrator.swift]: we should keep the headers as following 
// Copyright © 2018 INLOOPX. All rights reserved.

File: ImagePicker/UICollectionView+CellRegistrator.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//  Copyright © 2018 Inloop. All rights reserved.
+//
+
+import Foundation
+
+
```
Comment (radimhalfar) [inline ImagePicker/UICollectionView+CellRegistrator.swift]: xtra space

File: ImagePicker/UICollectionView+CellRegistrator.swift [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        registerCameraItem(registrator: registrator, cameraMode: cameraMode)
+        registerAssetItems(registrator: registrator)
+    }
+}
+
+// MARK: - Helper private methods
+private extension UICollectionView {
+    func registerActionItems(registrator: CellRegistrator) {
+        if registrator.hasUserRegisteredActionCell {
+            register(nibsData: registrator.actionItemNibsData?.map { $1 })
```
Comment (radimhalfar) [inline ImagePicker/UICollectionView+CellRegistrator.swift:21]: both mappings done using $1 ? Seems weird

File: ImagePicker/UICollectionView+CellRegistrator.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        }
+        let nib = UINib(nibName: "ActionCell", bundle: Bundle(for: ActionCell.self))
+        register(nib, forCellWithReuseIdentifier: identifier)
+    }
+
+    func registerAssetItems(registrator: CellRegistrator) {
+        register(nibsData: registrator.assetItemNibsData?.map { $1 })
+        register(classData: registrator.assetItemClassesData?.map { $1 })
+        switch (registrator.assetItemNib, registrator.assetItemClass) {
+
```
Comment (radimhalfar) [inline ImagePicker/UICollectionView+CellRegistrator.swift]: xtra space

**Review by radimhalfar — CHANGES_REQUESTED** [review]

File: ImagePicker/UICollectionView+CellRegistrator.swift (hunk 4 above)
Comment (asirokova) [reply ImagePicker/UICollectionView+CellRegistrator.swift:21]: That's not my code, but as I can see, it's correct: `actionItemNibsData` and `actionItemClassesData` are dictionaries:
```swift
    var actionItemNibsData: [Int: (UINib, String)]?
    var actionItemClassesData: [Int: (UICollectionViewCell.Type, String)]?
```
and `register` methods are getting corresponding values, that's why there is $1

**Comment by asirokova** [comment]
@radimhalfar fixed :)

**Review by radimhalfar — APPROVED** [review]

**Pull request merged**